packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -126,13 +127,26 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.setup_params = setup_params
         super().__init__(*args, **kwargs)
 
+        if not self.context.is_abstract_component:
+            # setup data are mandatory for non-abstract skills,
+            # and they should always contain at least the `safe_contract_address`
```

Comment:
And `all_participants`. Please discuss with Viraj how we can configure CLI to directly pass both values in. For e2e tests we will have to have some changes to in order to properly populate this.

Reply:
We would simply need to add `all_participants` to the `setup` params in the config and create a command like `_try_update_safe_contract_address` for `all_participants`:

https://github.com/valory-xyz/open-autonomy/blob/6395ee932295b185597f64a3934eeae56c0ae978/autonomy/deploy/base.py#L229

Reply:
I suggest you do this in a new PR on top - it will affect a lot of tests

Reply:
Done in https://github.com/valory-xyz/open-autonomy/pull/1677.